src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx

```diff
@@ -163,97 +172,182 @@ class GroupDetails extends React.Component<Props, State> {
+      if (
+        reprocessingStatus === ReprocessingStatus.REPROCESSING &&
+        currentTab !== TAB.DETAILS
+      ) {
+        return {
+          pathname: baseUrl,
+          query: params,
+        };
+      }
+
+      if (
+        reprocessingStatus === ReprocessingStatus.REPROCESSED_AND_HASNT_EVENT &&
+        (currentTab !== TAB.ACTIVITY || currentTab !== TAB.USER_FEEDBACK)
+      ) {
+        return {
+          pathname: `${baseUrl}${TAB.ACTIVITY}/`,
+          query: params,
+        };
+      }
+    }
+
+    return undefined;
+  }
+
+  getGroupQuery(): Record<string, string | string[]> {
+    const {environments, organization} = this.props;
+
+    // Note, we do not want to include the environment key at all if there are no environments
+    const query: Record<string, string | string[]> = {
+      ...(environments ? {environment: environments} : {}),
+    };
+
+    if (organization?.features?.includes('inbox')) {
+      query.expand = 'inbox';
+    }
+
+    return query;
+  }
+
+  getFetchDataRequestErrorType(status: any): Error {
+    if (!status) {
+      return null;
+    }
+
+    if (status === 404) {
+      return ERROR_TYPES.GROUP_NOT_FOUND;
+    }
+
+    if (status === 403) {
+      return ERROR_TYPES.MISSING_MEMBERSHIP;
+    }
+
+    return null;
+  }
+
+  handleRequestError(error: any) {
+    Sentry.captureException(error);
+    const errorType = this.getFetchDataRequestErrorType(error?.status);
+
+    this.setState({
+      loadingGroup: false,
+      loading: false,
+      error: true,
+      errorType,
+    });
+  }
+
+  refetchGroup = async () => {
+    const {loadingGroup, loading, loadingEvent, group} = this.state;
+
+    if (
+      group?.status !== ReprocessingStatus.REPROCESSING ||
+      loadingGroup ||
+      loading ||
+      loadingEvent
+    ) {
+      return;
+    }
+
+    const {api} = this.props;
+    this.setState({loadingGroup: true});
+
+    try {
+      const updatedGroup = await api.requestPromise(this.groupDetailsEndpoint, {
+        query: this.getGroupQuery(),
+      });
+
+      const reprocessingNewRoute = this.getReprocessingNewRoute(updatedGroup);
+
+      if (reprocessingNewRoute) {
+        ReactRouter.browserHistory.push(reprocessingNewRoute);
+        return;
+      }
+
+      this.setState({group: updatedGroup, loadingGroup: false});
+    } catch (error) {
+      this.handleRequestError(error);
+    }
+  };
+
+  async fetchData() {
+    const {api, isGlobalSelectionReady, params} = this.props;
+
+    // Need to wait for global selection store to be ready before making request
+    if (!isGlobalSelectionReady) {
+      return;
+    }
+
+    try {
+      const eventPromise = this.canLoadEventEarly(this.props)
+        ? this.getEvent()
+        : undefined;
+
+      const groupPromise = await api.requestPromise(this.groupDetailsEndpoint, {
+        query: this.getGroupQuery(),
+      });
+
+      const [data] = await Promise.all([groupPromise, eventPromise]);
```

Comment:
`groupPromise` gets assigned to`data` variable, what about eventPromise? Is that just some sort of side-effect?

Reply:
oh I did not write this code and did not want to change it in this PR